### ISSUE TITLE
Cloudflare waiting room

### ISSUE THREAD

**Issue by AntonBorisovich** [realcoloride/node_characterai#56]
When I try to log in (both as a Guest and with a token), I get this error.
```js
{
  name: 'SyntaxError',
  message: "Unexpected '<'",
  at: 1,
  text: '<!DOCTYPE html>\n' +
    '<html>\n' +
    '  <head>\n' +
    '    <title>Waiting Room powered by Cloudflare</title>\n' +
    '    <meta name="viewport" content="width=device-width, initial-scale=1" />\n' +
    '    <style type="text/css">\n' +
    '      * {\n' +
    '        box-sizing: border-box;\n' +
    '      }\n' +
    '\n' +
    '      html,\n' +
    '      body,\n' +
    '      #wrapper {\n' +
    '        padding: 0;\n' +
    '        margin: 0;\n' +
    '        height: 100%;\n' +
    '      }\n' +
    '\n' +
    '      #wrapper {\n' +
    '        display: table;\n' +
    '        width: 100%;\n' +
    '      }\n' +
    '\n' +
    '      main,\n' +
    '      footer {\n' +
    '        /* Old browsers */\n' +
    '        display: block;\n' +
    '        /* Modern browsers */\n' +
    '        display: table-row;\n' +
    '      }\n' +
    '\n' +
    '      body {\n' +
    "        font-family: -apple-system, system-ui, BlinkMacSystemFont, 'Segoe UI',\n" +
    "          Roboto, Oxygen, Ubuntu, 'Helvetica Neue', Arial, sans-serif;\n" +
    '        font-size: 16px;\n' +
    '        margin: 0;\n' +
    '        padding: 0;\n' +
    '      }\n' +
    '\n' +
    '      main .content-container {\n' +
    '        max-width: 1024px;\n' +
    '      }\n' +
    '\n' +
    '      .center {\n' +
    '        height: 100%;\n' +
    '        display: flex;\n' +
    '        align-items: center;\n' +
    '      }\n' +
    '\n' +
    '      .content-container {\n' +
    '        width: 100%;\n' +
    '        min-width: 400px;\n' +
    '        margin: 0 auto;\n' +
    '        padding: 32px 64px;\n' +
    '        display: flex;\n' +
    '      }\n' +
    '\n' +
    '      @media (max-width: 900px) {\n' +
    '        .content-container {\n' +
    '          display: block;\n' +
    '        }\n' +
    '      }\n' +
    '\n' +
    '      main .content-container section {\n' +
    '        flex: 1 0 0;\n' +
    '        display: flex;\n' +
    '        flex-direction: column;\n' +
    '      }\n' +
    '\n' +
    '      footer {\n' +
    '        height: 100px;\n' +
    '        background: #138eed;\n' +
    '        color: white;\n' +
    '        font-size: 12px;\n' +
    '      }\n' +
    '\n' +
    '      footer .content-container {\n' +
    '        justify-content: space-between;\n' +
    '      }\n' +
    '\n' +
    '      p {\n' +
    '        line-height: 1.5;\n' +
    '        max-width: 34rem;\n' +
    '      }\n' +
    '\n' +
    '      h1 {\n' +
    '        font-size: 32px;\n' +
    '        margin: 0 0 32px 0;\n' +
    '      }\n' +
    '\n' +
    '      h2 {\n' +
    '        font-size: 20px;\n' +
    '        margin: 0 0 20px 0;\n' +
    '      }\n' +
    '\n' +
    '      .links {\n' +
    '        color: #cac8c8;\n' +
    '        margin-top: 15px;\n' +
    '        display: flex;\n' +
    '        align-items: flex-end;\n' +
    '      }\n' +
    '\n' +
    '      .links a,\n' +
    '      .links a:visited {\n' +
    '        text-decoration: none;\n' +
    '        color: #cac8c8;\n' +
    '      }\n' +
    '\n' +
    '      .logo {\n' +
    '        display: flex;\n' +
    '        align-items: flex-end;\n' +
    '      }\n' +
    '\n' +
    '      .logo .cloudflare-logo {\n' +
    '        width: 140px;\n' +
    '        margin-right: 1rem;\n' +
    '      }\n' +
    '    </style>\n' +
    '  </head>\n' +
    '\n' +
    '  <body>\n' +
    '    <div id="wrapper">\n' +
    '      <main>\n' +
    '        <div class="center">\n' +
    '          <section class="content-container">\n' +
    '\n' +
    '\n' +
    '            <section>\n' +
    '<img src="https://characterai.io/logo-light-bg.png" alt="logo" style="max-width: 400px;margin-left:-34px">\n' +
    '<br/>\n' +
    '              <h1>\n' +
    '                You are now in line.\n' +
    '                <br />\n' +
    '                Thanks for your patience.\n' +
    '              </h1>\n' +
    '            </section>\n' +
    '            <section>\n' +
    '\n' +
    '              <h2>\n' +
    '                        Your estimated wait time is 2 minutes...\n' +
    `            <a href="https://plus.character.ai" style="text-decoration:none;"><div style='background-color: #000000 ;width: 100%; padding: 10px 30px;color: white;display: flex;justify-content: center; align-items: center;font-size: 20px;border-radius: 16px; cursor: pointer;margin-top: 16px;'>c.ai+ user? Click here to skip the line!</div></a>\n` +
    '            <br >\n' +
    `            <a href="https://plus.character.ai" target="_blank" style="text-decoration:none;"><div style='background-color: #056DFF; width: 100%; padding: 10px 30px;color: white;display: flex;justify-content: center; align-items: center;font-size: 20px;border-radius: 16px; cursor: pointer;'>Get c.ai+</div></a>\n` +
    '              </h2>\n' +
    '              <p>\n' +
    '                We are experiencing a high volume of traffic and using a virtual\n' +
    '                queue to limit the number of users on Character.AI at the same\n' +
    '                time.\n' +
    '              </p>\n' +
    '             <p>\n' +
    "                 Your Characters will be waiting for you when it's your turn!\n" +
    '             </p>\n' +
    '              <p>\n' +
    '                <strong>\n' +
    '                    This page will automatically refresh, please do not close your\n' +
    '                    browser.\n' +
    '                </strong>\n' +
    '              </p>\n' +
    '              <p>\n' +
    '                <strong>\n' +
    '                  Last updated\n' +
    '                  <span id="last-updated"></span>\n' +
    '                </strong>\n' +
    '              </p>\n' +
    '            </section>\n' +
    '          </section>\n' +
    '        </div>\n' +
    '      </main>\n' +
    '    </div>\n' +
    '    <script>\n' +
    '      var date = new Date();\n' +
    "      document.getElementById('year').innerText = date.getFullYear();\n" +
    '      document.getElementById(\n' +
    "        'last-updated'\n" +
    '      ).innerText = date.toLocaleTimeString();\n' +
    '    </script>\n' +
    `  <script>(function(){var js = "window['__CF$cv$params']={r:'TOKEN'};_cpo=document.createElement('script');_cpo.nonce='',_cpo.src='/cdn-cgi/challenge-platform/scripts/invisible.js',document.getElementsByTagName('head')[0].appendChild(_cpo);";var _0xh = document.createElement('iframe');_0xh.height = 1;_0xh.width = 1;_0xh.style.position = 'absolute';_0xh.style.top = 0;_0xh.style.left = 0;_0xh.style.border = 'none';_0xh.style.visibility = 'hidden';document.body.appendChild(_0xh);function handler() {var _0xi = _0xh.contentDocument || _0xh.contentWindow.document;if (_0xi) {var _0xj = _0xi.createElement('script');_0xj.nonce = '';_0xj.innerHTML = js;_0xi.getElementsByTagName('head')[0].appendChild(_0xj);}}if (document.readyState !== 'loading') {handler();} else if (window.addEventListener) {document.addEventListener('DOMContentLoaded', handler);} else {var prev = document.onreadystatechange || function () {};document.onreadystatechange = function (e) {prev(e);if (document.readyState !== 'loading') {document.onreadystatechange = prev;handler();}};}})();</script><script defer src="https://static.cloudflareinsights.com/beacon.min.js/TOKEN" integrity="sha512-TOKEN" data-cf-beacon='{"rayId":"TOKEN","token":"TOKEN","version":"2023.4.0","si":100}' crossorigin="anonymous"></script>\n` +
    '</body>\n' +
    '</html>'
}
```
At the end I removed the lines that could be tokens and replaced them with word "TOKEN", just in case
Currently using 1.1.4, but with 1.1.3 same error

**Comment by Parking-Master** [comment]
If you go on the Character.AI website, you should also get that message. This is for the users who don't have c.ai+ accounts or are accessing as guest. You will need to wait a few minutes for it to go away.

For some reason the AI website is receiving a truckload of requests today, I don't remember them having this many requests. Just try waiting and try again later. Wish you the best of luck :D

**Comment by AntonBorisovich** [comment]
I just read changelog for v1.1.4 and thought that you can somehow wait queue. Also after 2 hours i still get this error.
![image](https://github.com/realcoloride/node_characterai/assets/36966367/5f364a96-4d7a-43fe-a856-2972f8ec7f1a)

**Comment by realcoloride** [comment]
I'm sorry that the new solution to wait in the waiting room didn't work. I will investigate the issue, meanwhile, make a try{} catch statement for now.

I'll keep you updated

**Comment by Parking-Master** [comment]
@realcoloride thanks! I am also getting this error too now.

**Comment by realcoloride** [comment]
Hello,
This issue might have a potential link to issue #64, so stay up to date.

**Comment by Parking-Master** [comment]
> This issue might have a potential link to issue https://github.com/realcoloride/node_characterai/issues/64, so stay up to date.

This issue can be closed, since it was related to #64 and it's been resolved.
